src/storage/hp/primera/restapi/custom/api.pm

```diff
@@ -175,6 +175,21 @@ sub request_api {
         critical_status => ''
     );
 
+    # Maybe token is invalid. so we retry
```

Comment:
This condition is equivalent to 
```
    if (
          ( !defined($self->{token}) && $self->{http}->get_code() < 200 )
          || $self->{http}->get_code() >= 300) {
```
which is probably not what you meant.

I suggest 

```
    if (!defined($self->{token}) || $self->{http}->get_code() >= 400) {
```
which stands for `if no token or if unauthorized`

Reply:
yes you are right ! this is a typo .. the && should be a ||, too.

if no token or >= 400 is fine for me. thanks